### ISSUE TITLE
Typescript errors not being thrown

### ISSUE THREAD

**Issue by mfmontenegro** [evanw/esbuild#1631]
Hi, first of all I would like to thank @evanw for the effort he puts on this project. 

I'm finding that **esbuild** shallows typescript errors. Walked through the documentation and couldn't see anything related. [Log level param](https://esbuild.github.io/api/#log-level) won't make any effect on this.

**node version**: 14.17.5
**npm version**: 7.24.0
**esbuild version**: 0.13.1

Example script:
`esbuild src/index.ts --bundle --outfile=public/index.js`

Example code:
```
  const test: number = 'An extremely fast JavaScript bundler'
  console.log(test)
```

Maybe am I missing out something? Is this a known Issue? Didn't find anything related on the Issues either.
Thanks!

**Comment by evanw** [comment]
From [esbuild's documentation on TypeScript](https://esbuild.github.io/content-types/#typescript):

> However, esbuild _does not_ do any type checking so you will still need to run `tsc -noEmit` in parallel with esbuild to check types. This is not something esbuild does itself.

**Comment by evanw** [comment]
Replicating the TypeScript type system is out of scope. Closing.

**Comment by mfmontenegro** [comment]
Thank you
